phoenix-core/src/it/java/org/apache/phoenix/end2end/InListIT.java

```diff
@@ -537,4 +539,267 @@ public void testInListExpressionWithDesc() throws Exception {
+            assertTrue(rs.next());
+            assertEquals("000000000000400", rs.getString(1));
+        }
+    }
+
+    private String getExplainPlanString(ResultSet rs) throws SQLException {
+        StringBuilder builder = new StringBuilder();
+        while (rs.next()) {
+            for (int i = 0; i < rs.getMetaData().getColumnCount(); i++) {
+                Object col = rs.getObject(i + 1);
+                if (col != null) {
+                    builder.append(col.toString());
+                    builder.append(",");
+                }
+            }
+            builder.append("\n");
+        }
+        return builder.toString();
+    }
+
+    @Test
+    public void testInListExpressionGeneratesRightScan() throws Exception {
+        String fullTableName = generateUniqueName();
+        String fullViewName = generateUniqueName();
+        String tenantView = generateUniqueName();
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            conn.setAutoCommit(true); Statement stmt = conn.createStatement();
+            stmt.execute("CREATE TABLE " + fullTableName + "(\n" + " TENANT_ID CHAR(15) NOT NULL,\n" + " KEY_PREFIX CHAR(3) NOT NULL,\n" +
+                    " CONSTRAINT PK PRIMARY KEY (\n" + " TENANT_ID," + " KEY_PREFIX" + ")) MULTI_TENANT=TRUE");
+            stmt.execute("CREATE VIEW " + fullViewName + "(\n" + " ID1 VARCHAR NOT NULL,\n" + " ID2 VARCHAR NOT NULL,\n" + " ID3 BIGINT, ID4 BIGINT \n" +
+                    " CONSTRAINT PKVIEW PRIMARY KEY\n" + " (ID1, ID2)) " +
+                    "AS SELECT * FROM " + fullTableName + " WHERE KEY_PREFIX = '0CY'");
+        }
+        try (Connection viewConn = DriverManager.getConnection(TENANT_SPECIFIC_URL1) ) {
+            viewConn.setAutoCommit(true);
+            Statement stmt = viewConn.createStatement();
+            stmt.execute("CREATE VIEW IF NOT EXISTS " + tenantView + " AS SELECT * FROM " + fullViewName);
+        }
+        testFullPkListPlan(tenantView);
+        testFullPkPlusNonPkInListPlan(tenantView);
+        testPartialPkListPlan(tenantView);
+        testPartialPkPlusNonPkListPlan(tenantView);
+        testNonPkListPlan(tenantView);
+    }
+
+    private void testFullPkPlusNonPkInListPlan(String tenantView) throws Exception {
+        Long numberOfRowsToScan = new Long(2);
+        try (Connection viewConn = DriverManager.getConnection(TENANT_SPECIFIC_URL1) ) {
+            PreparedStatement preparedStmt = viewConn.prepareStatement("SELECT * FROM " + tenantView + " WHERE (ID1, ID2, ID3) IN " +
+                    "(('005xx000001Sv6o', '000000000000500', 1)," +
+                    "('005xx000001Sv6o', '000000000000400', 2))");
+            QueryPlan queryPlan = PhoenixRuntime.getOptimizedQueryPlan(preparedStmt);
+            assertEquals(numberOfRowsToScan, queryPlan.getEstimatedRowsToScan());
+            assertTrue(queryPlan.getExplainPlan().toString().contains(ExplainTable.POINT_LOOKUP_ON_STRING));
```

Comment:
Any reason why we only extracted `POINT_LOOKUP_ON_STRING` string out in `ExplainTable` and not any of the other ones?

Reply:
other strings are not easy to abstract due to the way how to generates the diff query plan